### ISSUE TITLE
FR: Caption Text Within Images

### ISSUE THREAD

**Issue by ZoeBeykirch** [hlxsites/vg-macktrucks-com#745]
### Context/Description

As a Mack Trucks content editor, I want to add caption text within an image block that can contain 1, 2, or 4 images so that I can provide context and descriptions for each image in a visually organized manner to enhance my magazine and news articles. 

### Acceptance Criteria

**User AC:**

- [ ] Users can view 1, 2 or 4 images with caption text overlaid on each image in the grid.
- [ ] The text color is white with a boxed background color of graphite. 

**Editor AC:**
- [ ]  As an editor I can choose to display 1, 2, or 4 images in this block. 
- [ ]  As an editor I can write the captions without a word limit. 
- [ ]  As an editor I can use superscript within the captions. 

**Non-functional AC:**
- [ ] Create variant of existing block [V2 Images grid] that shall be named  V2 Image grid (with captions)
- [ ] This will be a variant without the existing modal functionality
- [ ] Block can be used in the Magazine and News Article Templates. 
- [ ]  Block can be used on any v2 redesign page. 
- [ ]  Superscript and bold font are supported. 
- [ ] If the caption text is longer than the width of the image, it will wrap to the next line. 
- [ ] Text will wrap up to two lines

**Out of Scope**
This is related to images only and does not need to overlay videos as a requirement

### Attachment / link to Figma

https://www.figma.com/design/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?m=dev&node-id=14119-7953

![Image](https://github.com/hlxsites/vg-macktrucks-com/assets/125895595/4cd8c937-a44c-4f89-bea5-7d2fe8a95b62)

**Comment by mollyaljenkins** [comment]
Waiting on updated design

**Comment by AsmitKMishra** [comment]
Testing Completed:
1. Editor has options to display 1,2 or 4 images.
2. Caption has no word limit and captions longer than image wraps in two lines.
3. Caption is limited to two lines.
4. Font and specs match with Figma.
